bin/crunchy-exporter/common_lib.sh

```diff
@@ -0,0 +1,172 @@
+    then
+        echo_warn "$1 environment variable is not set."
+    fi
+}
+
+function env_check_info() {
+    if [[ ! -z ${!1} ]]
+    then
+        echo_info "$2"
+    fi
+}
+
+function dir_check_err() {
+    if [[ ! -d ${1} ]]
+    then
+        echo_err "The $1 directory does not exist and is required."
+        exit 1
+    fi
+}
+
+function echo_err() {
+    echo -e "${RED?}$(date) ERROR: ${1?}${RESET?}"
+}
+
+function echo_info() {
+    echo -e "${GREEN?}$(date) INFO: ${1?}${RESET?}"
+}
+
+function echo_warn() {
+    echo -e "${YELLOW?}$(date) WARN: ${1?}${RESET?}"
+}
+
+function pgisready() {
+    export PGROOT=$(find /usr/ -type d -name 'pgsql-*')
+    local dbname=${1?}
+    local dbhost=${2?}
+    local dbport=${3?}
+    local dbuser=${4?}
+    local max_attempts=${5:-5}
+    local timeout=${6:-2}
+    test_server ${dbname?} ${dbhost?} ${dbport?} ${dbuser?} ${max_attempts?} ${timeout?}
+    test_query ${dbname?} ${dbhost?} ${dbport?} ${dbuser?} ${max_attempts?} ${timeout?}
+}
+
+# Check if PostgreSQL is ready with exponential backoffs on attempts
+function test_server() {
+    local dbname=${1?}
+    local dbhost=${2?}
+    local dbport=${3?}
+    local dbuser=${4?}
+    local max_attempts=${5:-5}
+    local timeout=${6:-2}
+    local attempt=0
+    local error='false'
+
+    echo_info "Waiting for PostgreSQL to be ready.."
+    while [[ ${attempt?} < ${max_attempts?} ]]
+    do
+        ${PGROOT?}/bin/pg_isready \
+            --dbname=${dbname?} --host=${dbhost?} \
+            --port=${dbport?} --username=${dbuser?}
+        if [[ $? -eq 0 ]]
+        then
+            error='false'
+            break
+        fi
+        error='true'
+        sleep ${timeout?}
+        attempt=$(( attempt + 1 ))
+        timeout=$(( timeout * 2 ))
+    done
+
+    if [[ ${error?} == 'true' ]]
+    then
+        echo_err "Could not connect to PostgreSQL: Host=${dbhost?}:${dbport?} DB=${dbname?} User=${dbuser?}"
+        exit 1
+    fi
+}
+
+# Check if PostgreSQL is ready with exponential backoffs on attempts
+function test_query {
+    local dbname=${1?}
+    local dbhost=${2?}
+    local dbport=${3?}
+    local dbuser=${4?}
+    local max_attempts=${5:-5}
+    local timeout=${6:-2}
+    local attempt=0
+    local error='false'
+
+    echo_info "Checking if PostgreSQL is accepting queries.."
+    while [[ ${attempt?} < ${max_attempts?} ]]
+    do
+        ${PGROOT?}/bin/psql \
+            --dbname=${dbname?} --host=${dbhost?} \
+            --port=${dbport?} --username=${dbuser?} \
+            --command="SELECT now();"
+        if [[ $? -eq 0 ]]
+        then
+            error='false'
+            break
+        fi
+        error='true'
+        sleep ${timeout?}
+        attempt=$(( attempt + 1 ))
+        timeout=$(( timeout * 2 ))
+    done
+
+    if [[ ${error?} == 'true' ]]
+    then
+        echo_err "Could not run query against PostgreSQL: Host=${dbhost?}:${dbport?} DB=${dbname?} User=${dbuser?}"
+        exit 1
+    fi
+}
```

Comment:
I think we can remove this.

Reply:
Sure thing, updated.

Reply:
On further review, there were quite a few functions that could be cleared out with the new changes, so I've removed those as well.